### PR TITLE
feat(app): add new config value to store sortKey for protocols

### PR DESCRIPTION
**Pull request by koji**
# Overview
This PR adds a new config value to store the sort key that a user used last.
Close RAUT-227

https://user-images.githubusercontent.com/474225/204846809-ce2bd0b1-b59e-4e35-8169-f30f2f5bf175.mov

# Changelog
- add a new config value as v11
- update test case

# Review requests
- go to the Protocols landing page and change the sort key then close the app. When opening the app, the protocols landing page sort key is the same as the key selected before closing the app.

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] fix test cases

File: app-shell/src/config/migrate.ts [hunk 1]
```diff
@@ -227,6 +228,19 @@ const toVersion10 = (prevConfig: ConfigV9): ConfigV10 => {
...
 }
 
+// config version 11 migration and defaults
+const toVersion11 = (prevConfig: ConfigV10): ConfigV11 => {
+  const nextConfig = {
+    ...prevConfig,
+    version: 11 as const,
+    protocolsSortKey: {
+      protocolsStoredSortKey: null,
+    },
```
Comment (brenthagen) [inline app-shell/src/config/migrate.ts]: i think it makes sense to put this under the `protocols` key created in v10:
```
protocols: {
  ...prevConfig.protocols,
  protocolsStoredSortKey: null,
}
```

File: app/src/organisms/ProtocolsLanding/ProtocolList.tsx [hunk 2]
```diff
@@ -47,8 +53,11 @@ interface ProtocolListProps {
   storedProtocols: StoredProtocolData[]
 }
 export function ProtocolList(props: ProtocolListProps): JSX.Element | null {
+  const protocolsStoredSortKey = useSelector(getProtocolsStoredSortKey)
   const [showSlideout, setShowSlideout] = React.useState<boolean>(false)
-  const [sortBy, setSortBy] = React.useState<ProtocolSort>('alphabetical')
+  const [sortBy, setSortBy] = React.useState<ProtocolSort>(
+    protocolsStoredSortKey ?? 'alphabetical'
+  )
```
Comment (brenthagen) [inline app/src/organisms/ProtocolsLanding/ProtocolList.tsx]: can we avoid deriving local state from redux and just read directly from redux to set the key?
```
const sortBy = useSelector(getProtocolsStoredSortKey) ?? 'alphabetical'
```

**Comment by koji** [comment]
- [x] address comments
  - [x] update test

**Review by brenthagen — APPROVED** [review]
did the font file get in this PR by mistake? otherwise looks good ✔️

**Comment by koji** [comment]
> did the font file get in this PR by mistake? otherwise looks good ✔️

My bad, yes, that file shouldn't be in this PR. Thank you for catching this!
I will remove it.

**Pull request merged**